yt_dlp/extractor/facebook.py

```diff
@@ -420,6 +420,30 @@ def extract_metadata(webpage):
                 r'data-sjs>({.*?ScheduledServerJS.*?})</script>', webpage)]
             post = traverse_obj(post_data, (
                 ..., 'require', ..., ..., ..., '__bbox', 'require', ..., ..., ..., '__bbox', 'result', 'data'), expected_type=dict) or []
+            snippet = traverse_obj(post, (..., 'video', ..., 'attachments', ..., lambda k, v: (
+                k == 'media' and str(v['id']) == video_id and v['__typename'] == 'Video')), expected_type=dict) or {}
+            locale = self._html_search_meta(['og:locale', 'twitter:locale'], webpage, 'locale', default='en_US')
+            captions = get_first(snippet, 'video_available_captions_locales', 'captions_url')
+            useIsVideoBroadcast = get_first(snippet, ('is_video_broadcast')) or False
+            automatic_captions = {}
+            subtitles = {}
+            if isinstance(captions, str):
+                subtitles[locale] = [{'ext': determine_ext(captions, default_ext='srt'), 'url': captions}]
+            elif isinstance(captions, list):
+                if len(captions) > 1:
+                    captions = sorted(captions, key=lambda c: (c['locale'] != locale, c['locale']))
+                for c in captions:
+                    s = {
+                        'ext': determine_ext(c['captions_url'], default_ext='srt'),
+                        'url': c['captions_url'],
+                        'name': (c['localized_language']
+                                 + (' (' + c['localized_country'] + ')' if c['localized_country'] else '')
+                                 + (' (' + c['localized_creation_method'] + ')' if c['localized_creation_method'] else '')),
+                    }
+                    if c['localized_creation_method'] or useIsVideoBroadcast:
+                        automatic_captions.setdefault(c['locale'], []).append(s)
+                    else:
+                        subtitles.setdefault(c['locale'], []).append(s)
```

Comment:
I used parentheses because it is common practice to present language & locale (e.g. YouTube, language setting in os).
Of course, we can use something else. To simplify the name, I think 'localized_creation_method' can be removed because we already distinguish between automatic captions and subtitles.

without sorting, we will get someting like this. imo, it is not user-friendly and messy, especially when two or more exist (e.g. English, 中文, Español, etc.)
<pre>
[facebook] Extracting URL: https://www.facebook.com/facebook/videos/10157064900956729/
[facebook] 10157064900956729: Downloading webpage
10157064900956729 has no automatic captions
[info] Available subtitles for 10157064900956729:
Language Name                             Formats
en_US    English-United States of America srt
zh_HK    中文-香港                            srt
zh_CN    中文-中国                            srt
vi_VN    Tiếng Việt                       srt
zh_TW    中文-台灣                            srt
tr_TR    Türkçe                           srt
tl_PH    Filipino                         srt
th_TH    ภาษาไทย                          srt
ur_PK    اردو                             srt
te_IN    తెలుగు                           srt
sk_SK    Slovenčina                       srt
sw_KE    Kiswahili                        srt
sr_RS    српски                           srt
sv_SE    Svenska                          srt
ta_IN    தமிழ்                            srt
pt_BR    Português-Brasil                 srt
pl_PL    polski                           srt
pt_PT    Português-Portugal               srt
ru_RU    Русский                          srt
ro_RO    Română                           srt
nl_NL    Nederlands                       srt
ms_MY    Bahasa Melayu                    srt
nb_NO    Norsk (bokmål)                   srt
pa_IN    ਪੰਜਾਬੀ                           srt
mr_IN    मराठी                            srt
ml_IN    മലയാളം                           srt
ja_JP    日本語                              srt
ko_KR    한국어                              srt
kn_IN    ಕನ್ನಡ                            srt
id_ID    Bahasa Indonesia                 srt
hi_IN    हिन्दी                           srt
hr_HR    hrvatski                         srt
it_IT    Italiano                         srt
hu_HU    magyar                           srt
gu_IN    ગુજરાતી                          srt
fi_FI    suomi                            srt
fr_FR    Français                         srt
es_LA    Español-Laos                     srt
he_IL    עברית                            srt
el_GR    Ελληνικά                         srt
de_DE    Deutsch                          srt
en_GB    English-United Kingdom           srt
da_DK    Dansk                            srt
es_ES    Español-España                   srt
ar_AR    العربية                          srt
bn_IN    বাংলা                            srt
af_ZA    Afrikaans                        srt
bg_BG    български                        srt
cs_CZ    Čeština                          srt
</pre>